coderd/activitybump.go

```diff
@@ -0,0 +1,77 @@
+package coderd
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/coderd/database"
+)
+
+// activityBumpWorkspace automatically bumps the workspace's auto-off timer
+// if it is set to expire soon.
+func activityBumpWorkspace(log slog.Logger, db database.Store, workspace database.Workspace) {
+	// We set a short timeout so if the app is under load, these
+	// low priority operations fail first.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
```

Comment:
I don't think its super unlikely for us to hit this timeout if there are some slightly longer running transactions active. We could consider raising it a tad bit.